### ISSUE TITLE
Can the editable install come with `package data`? i.e. `pip install -e .`

### ISSUE THREAD

**Issue by noklam** [kedro-org/kedro-viz#1611]
## Description

Currently, when `kedro-viz` is installed locally, the assets such as `html` and `css` are missing. The contribution guide say that you need to run `make build`. 

This requires the user has `npm` and make it harder to contribute particularly if one is interested in the Python bit but not the front-end.

Is is possible to make it work without a separate build step? Cc @astrojuanlu 

I found some links in setuptools:
- Manifest.in - https://setuptools.pypa.io/en/latest/userguide/miscellaneous.html#controlling-files-in-the-distribution
- https://setuptools.pypa.io/en/latest/userguide/datafiles.html

## Checklist

- [x] Include labels so that we can categorise your feature request

**Comment by astrojuanlu** [comment]
Yes it can be done: https://github.com/kedro-org/kedro-viz/issues/1267#issuecomment-1565240967

**Comment by ravi-kumar-pilla** [comment]
@astrojuanlu Thank you for the comment. Kedro Viz sdist already contains the js build. But, this ticket is to include js build in local development I guess. We do not have our js build on Git and if someone wants to contribute `only python` code still needs the js build as the html folder is missing from the version control. 

I do not see how the Manifest.in or the datafiles link you shared might help this usecase @noklam , as they are helpful in including the files which are already in the source tree. Please correct me if I am wrong. 

To resolve this issue, we need to somehow include the html folder (js build) in the source tree (VCS), so that users can get the files for local development. Happy to hear any other way to do this. Thank you

**Comment by rashidakanchwala** [comment]
Closing this issue as it cannot be done

**Comment by rashidakanchwala** [comment]
@astrojuanlu - Requested to reopen this issue. Maybe we can invite you to our next backlog grooming and u can give us more information on this and #1267

**Comment by astrojuanlu** [comment]
I'm happy to provide a PR for it after the codebase is migrated to a non-obsolete version of Node.js (so that I am not blocked by https://github.com/kedro-org/kedro-viz/issues/1278)

![image](https://github.com/kedro-org/kedro-viz/assets/316517/99b953e8-9969-44b7-940a-b26564c8fa5d)

Version 16.x is EOL since September 2023.

**Comment by astrojuanlu** [comment]
Now that the codebase has been migrated to Node.js 18 #1652 this should be easier to do.

**Comment by astrojuanlu** [comment]
Explanation (and fix) of #1267 in https://github.com/kedro-org/kedro-viz/pull/1766

**Comment by antonymilne** [comment]
This is something I've always wanted to have! So +💯 from me if it can be done 😁

**Comment by ravi-kumar-pilla** [comment]
This is a developer and contributor experience enhancement task. We can -

1. **Commit built HTML** - lowest friction, highest maintenance cost on frontend PRs.
2. **Hatch build hook** - keeps HTML out of git, but npm is still required once at install time.

`MANIFEST.in` / hatch `package_data` alone will not fix this unless the HTML already exists in the tree (option 1) or a hook generates it (option 2). 

My recommendation would be to close this as not planned. Thank you

**Comment by ElenaKhaustova** [comment]
Based on our discussion with the team, we've decided to close this ticket as it only affects developer experience which is clarified in the contribution guidelines.
